### PR TITLE
Generate strongly typed On%s and On%sMatch methods

### PR DESCRIPTION
**Pull request by iaroslav-ciupin**
Description
-------------

This PR is a follow up on https://github.com/vektra/mockery/pull/247. Original author didn't have time to address comments and finish it. I'm applying same change on top of current master.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

Version of Golang used when building/testing:
---------------------------------------------

- [x] 1.19

How Has This Been Tested?
---------------------------

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Checklist
-----------

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by iaroslav-ciupin** [comment]
@LandonTClipp Hi, this a re-iteration of the old PR https://github.com/vektra/mockery/pull/247. Are you OK with implemented approach? if yes, I will fix all unit tests and make PR ready for review.

File: pkg/generator.go [hunk 1]
```diff
@@ -629,6 +630,54 @@ func (g *Generator) Generate(ctx context.Context) error {
...
+				g.getInstantiatedTypeString(),
+				fname,
+				callType,
+				g.getInstantiatedTypeString(),
+			)
+			{
+				g.printf("\tc := _m.On(\"%s\", matchers...)\n", fname)
+				g.printf("\treturn &%s%s{Call: c}\n", callType, g.getInstantiatedTypeString())
+			}
+			g.printf("}\n")
```
Comment (LandonTClipp) [inline pkg/generator.go:677]: Sorry to create more work, but you could change this to using `text/template`? The individual prints are kind of hard to read, I've found templates to be more understandable.

**Comment by LandonTClipp** [comment]
I am in favor of this approach, I think this is the same general idea as the original PR. Is there something substantial difference here?

**Comment by iaroslav-ciupin** [comment]
@LandonTClipp sorry, I just realized that the desired functionality is already there. What I was looking for is `--with-expecter` flag and `mock.EXPECT()` syntax. I now think that maybe I don't need this change anymore and PR can be closed.

**Pull request closed without merging**